lib/goose/templates/migration-main.go.tmpl

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"log"
+	"bytes"
+	"encoding/gob"
+
+	_ "{{.Import}}"
+	"github.com/CloudCom/goose/lib/goose"
+)
+
+func main() {
+
+	var conf goose.DBConf
+	buf := bytes.NewBuffer({{ .Conf }})
+	if err := gob.NewDecoder(buf).Decode(&conf); err != nil {
+		log.Fatal("gob.Decode - ", err)
+	}
+
+	db, err := goose.OpenDBFromDBConf(&conf)
+	if err != nil {
+		log.Fatal("failed to open DB:", err)
+	}
+	defer db.Close()
+
+	txn, err := db.Begin()
+	if err != nil {
+		log.Fatal("db.Begin:", err)
+	}
+
+	{{ .Func }}(txn)
+
+	err = goose.FinalizeMigration(&conf, txn, {{ .Direction }}, {{ .Version }})
+	if err != nil {
+		log.Fatal("Commit() failed:", err)
+	}
+}
+{{/* vim: set ft=go.gotexttmpl: */}}
```

Comment:
It's a vim modeline. Sets settings in vim when the file is opened. In this case it tells vim this is a go file within a go text template (syntax highlighting, code formatting, etc).